sdk/test-utils/recorder/src/baseRecorder.ts

```diff
@@ -0,0 +1,389 @@
+      .replace(/<=/g, "lte")
+      .replace(/>=/g, "gte")
+      .replace(/</g, "lt")
+      .replace(/>/g, "gt")
+      .replace(/=/g, "eq")
+      .replace(/\W/g, "");
+  }
+
+  /**
+   * Additional layer of security to avoid unintended/accidental occurrences of secrets in the recordings
+   * */
+  protected filterSecrets(recording: string): string {
+    let updatedRecording = recording;
+    for (const k of Object.keys(replaceableVariables)) {
+      const escaped = escapeRegExp(env[k]);
+      updatedRecording = updatedRecording.replace(
+        new RegExp(escaped, "g"),
+        replaceableVariables[k]
+      );
+    }
+    for (const map of replacements) {
+      updatedRecording = map(updatedRecording);
+    }
+
+    // Handling storage environment variables separately
+    updatedRecording = recording.replace(new RegExp(env.ACCOUNT_NAME, "g"), "fakestorageaccount");
+    if (env.ACCOUNT_KEY) {
+      updatedRecording = updatedRecording.replace(new RegExp(env.ACCOUNT_KEY, "g"), "aaaaa");
+    }
+    if (env.ACCOUNT_SAS) {
+      updatedRecording = updatedRecording.replace(
+        new RegExp(env.ACCOUNT_SAS.match("(.*)&sig=(.*)")[2], "g"),
+        "aaaaa"
+      );
+    }
+    return updatedRecording;
+  }
+
+  public skip(): boolean {
+    return skip.includes(this.filepath);
+  }
+
+  public abstract record(): void;
+  public abstract playback(filePath: string): void;
+  public abstract stop(): void;
+}
+
+export class NockRecorder extends BaseRecorder {
+  constructor(testHierarchy: string, testTitle: string) {
+    super("node", testHierarchy, testTitle, "js");
+  }
+
+  public record(): void {
+    nock.recorder.rec({
+      dont_print: true
+    });
+  }
+
+  public playback(filePath: string): void {
+    let path = require("path");
+    this.uniqueTestInfo = require(path.resolve(
+      filePath,
+      "../../recordings/" + this.filepath
+    )).testInfo;
+  }
+
+  public stop(): void {
+    const importNock = "let nock = require('nock');\n";
+    const fixtures = nock.recorder.play();
+
+    // Create the directories recursively incase they don't exist
+    try {
+      // Stripping away the filename from the filepath and retaining the directory structure
+      fs.ensureDirSync(
+        "./recordings/" + this.filepath.substring(0, this.filepath.lastIndexOf("/") + 1)
+      );
+    } catch (err) {
+      if (err.code !== "EEXIST") throw err;
+    }
+
+    const file = fs.createWriteStream("./recordings/" + this.filepath, {
+      flags: "w"
+    });
+
+    // Some tests expect errors to happen and, if a writing error is thrown in one of these tests, it may be captured in a catch block by accident,
+    // resulting in unexpected behavior. For this reason we're printing it to the console as well
+    file.on("error", (err: any) => {
+      console.log(err);
+      throw err;
+    });
+
+    file.write(
+      importNock + "\n" + "module.exports.testInfo = " + JSON.stringify(this.uniqueTestInfo) + "\n"
```

Comment:
Move this entire string manipulation into the `importNock` constant and rename it to `importNockStatement`

Reply:
Done